R/dependencies.R

```diff
@@ -23,3 +23,30 @@ find_dependencies <- function(expr) {
        variables = unique(variables$get()))
 
 }
+
+
+topological_order <- function(deps) {
+  if (all(lengths(deps) == 0)) {
+    return(list(success = TRUE, value = seq_along(deps)))
+  }
+
```

Comment:
convert deps into a matrix of ndeps * ndeps - where m[x,y] - being TRUE implies the variable at index x is depended on, by the variable at index y. So in the test example:

a = c+1, b = a*2, c = b/2:

deps contanins $a = "c", $b = "a", $c = "b"

m[3,1] is true, (c is depended on by a), as are m[1,2] and m[2,3]